addons/website/tests/test_ui.py

```diff
@@ -317,3 +317,36 @@ def test_29_website_backend_menus_redirect(self):
         self.env.ref('base.user_admin').action_id = self.env.ref('base.menu_administration').id
         self.assertFalse(menu_root.action, 'The top menu should not have an action (or the test/tour will not test anything).')
         self.start_tour('/', 'website_backend_menus_redirect', login='admin')
+
+    def test_website_no_dirty_page(self):
+        # Having enough menus to trigger the "+" folded menus is known to
+        # wrongfully mark the page as dirty. So what should be done here is
+        # creating a lot of menus first. Actually there are 3 cases:
+        #
+        # - the menu is not folded outside of edit mode and when entering edit
+        #   mode, the "+" appears and some menu are folded
+        #
+        # - the menu is folded outside of edit mode and when entering edit mode
+        #   the resize actually makes it so a different set of menu is folded
+        #
+        # - the menu is folded outside of edit mode and when entering edit mode
+        #   it stays the same (known to have been broken because edit mode
+        #   tweaks the dropdown behavior)
```

Comment:
Arguably, could make a test for each of those + what I did here (testing when there is no "+" menu at all). Would be easy. What do you think @rdeodoo ?

Reply:
The more test the better I'd say, it's already very nice to have tackled those dirty behaviors (it's always a pain but I never really bother because I thought real client probably do not discard as often as us (if at all)), but if you add more, it's even nicer :o

Reply:
At long term, this could be more that just avoiding the useless discard message, it is about not cowing for no reason (but for that, we'll need more than this PR). I'll add those tests.